src/extensionFunctions.ts

```diff
@@ -599,4 +601,46 @@ export namespace extensionFunctions {
 			console.log('[' + msg + ']');
 		}
 	}
+
+	export async function validateCommandIDsInSelectedFile(didactUri: vscode.Uri) : Promise<void> {
+		if (didactUri) {
+			await vscode.commands.executeCommand(START_DIDACT_COMMAND, didactUri);
+		}
+		if (DidactWebviewPanel.currentPanel) {
+			const commands : any[] = extensionFunctions.gatherAllCommandsLinks();
+			if (commands && commands.length > 0) {
+				sendTextToOutputChannel(`Starting validation...`);
+				let allOk = true;
+				for(let command of commands) {
+					// validate all commands we found
+					const parsedUrl = url.parse(command, true);
+					const query = parsedUrl.query;
+					if (query.commandId) {
+						const commandId = utils.getValue(query.commandId);
+						if (commandId) {
+							const vsCommands : string[] = await vscode.commands.getCommands(true);
```

Comment:
I feared that a call to this method would be time-costing as there is an await and it is a platform call. So I tried to put it outside of the loop.
results:
with 100 commands in the didact file:
- as-is: between 350 and 600ms
- outside of the loop: between 50 and 150ms

so not a huge difference but it might remain interesting to move the retrieval of the VS Code commands outside of the loop.

Reply:
Something to consider